### ISSUE TITLE
incorrect line number and missing module name for E0633 (and false-positive)

### ISSUE THREAD

**Issue by Kwasniok** [PyCQA/pylint#4961]
### Bug description

```python
# pylint: disable=C0114,W0104

from typing import NewType

MyTuple = NewType("T", tuple[float, float])
x = MyTuple((1.1, 2.2))
a, b = x

# NOTE: The error description will always refer to line 10 exactly and refer to a module without a name.
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:7:0: E0633: Attempting to unpack a non-sequence defined at line 10 of  (unpacking-non-sequence)

------------------------------------------------------------------
Your code has been rated at 1.67/10 (previous run: 1.67/10, +0.00)
```

### Expected behavior

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.2
Python 3.9.6 (default, Jun 30 2021, 10:22:16) 
[GCC 11.1.0]
```

### OS / Environment

```
cat /proc/version 
Linux version 5.13.10-arch1-1 (linux@archlinux) (gcc (GCC) 11.1.0, GNU ld (GNU Binutils) 2.36.1) #1 SMP PREEMPT Thu, 12 Aug 2021 21:59:14 +0000
```

**Comment by Kwasniok** [comment]
The false-positive is known in similar cases: see #4895

**Comment by Kwasniok** [comment]
The output for a setup as in #4803 gives a suitable line and module reference with the exact same setup and error number.
